### DISCUSSION TITLE
c++ sample

### DISCUSSION THREAD

**Discussion by emad008** [trustwallet/wallet-core#1817] [Q&A]
Hi. I'm having a issue using cpp sample for wallet core. Docker image is getting build successfully but I cant make cpp sample. Here are the commands I've used (repository is fully up to date):

docker build . --tag wallet-core-dev
(build successfully. I have tested this with ubuntu 18.04 and 20.04)
docker run -i -t wallet-core-dev /bin/bash
inside docker image:
./tools/generate-files
cmake -H. -Bbuild -DCMAKE_BUILD_TYPE=Debug
make -Cbuild
cd samples/cpp
cmake . -DWALLET_CORE=../../
make
on the last command I encounter a long list of errors but the last part is like this:

/wallet-core/src/Cardano/AddressV2.cpp:25: undefined reference to `std::invalid_argument::invalid_argument(char const*)'
/wallet-core/src/Cardano/AddressV2.cpp:29: undefined reference to `std::invalid_argument::invalid_argument(char const*)'
/wallet-core/src/Cardano/AddressV2.cpp:33: undefined reference to `std::invalid_argument::invalid_argument(char const*)'
/wallet-core/samples/cpp/../../build/libTrustWalletCore.a(AddressV2.cpp.o):/wallet-core/src/Cardano/AddressV2.cpp:39: more undefined references to `std::invalid_argument::invalid_argument(char const*)' follow
/wallet-core/samples/cpp/../../build/libprotobuf.a(stringpiece.cc.o): In function `google::protobuf::operator<<(std::ostream&, google::protobuf::StringPiece)':
/wallet-core/build/local/src/protobuf/protobuf-3.14.0/src/google/protobuf/stubs/stringpiece.cc:43: undefined reference to `std::ostream::write(char const*, long)'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [CMakeFiles/sample.dir/build.make:97: sample] Error 1
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/sample.dir/all] Error 2

**Comment by optout21** [comment]
Have you tried to build it directly, not inside Docker?
https://github.com/trustwallet/wallet-core/blob/master/samples/cpp/README.md

Docker image we do not officially support.
Docker building script/env may be out of date, we will check.

**Reply by emad008** [reply]
I'm on linux and It's little complicated to build wallet core directly. I've tried the python fork and surprisingly it worked. I've also wrote a docker file for python build using the same docker file on wallet core project. It seems to be a problem with sample builds (I couldn't use go sample either)

**Comment by optout21** [comment]
Sample app building fails, if selected compiler is not clang.  On Ubuntu/Linux, often gcc is picked by default.
I plan to add building of sample app to the CI build, so it is kept up to date.
See here:   https://github.com/trustwallet/wallet-core/pull/1831
